libr/core/disasm.c

```diff
@@ -3773,7 +3773,19 @@ static int myregwrite(RAnalEsil *esil, const char *name, ut64 *val) {
 		R_FREE (type);
 		if ((ds->printed_flag_addr == UT64_MAX || *val != ds->printed_flag_addr)
 		    && (ds->show_emu_strflag || !emu_str_printed)) {
-			RFlagItem *fi = r_flag_get_i (esil->anal->flb.f, *val);
+			const RList *flags = r_flag_get_list (esil->anal->flb.f, *val);
+			RFlagItem *fi = NULL;
+			RListIter *iter;
+			RFlagItem *item;
+			r_list_foreach_prev (flags, iter, item) {
+				if (!item->name) {
+					continue;
+				}
+				fi = item;
+				if (strncmp (item->name, "section_end.", 12) != 0) {
```

Comment:
the != is unnecessary

Reply:
maybe it will be faster to check if fi-> have section_end. and then use r_flag_get_i2()

but i think we want to have a r_flag_get_except() or something

Reply:
i think faster still is having a separate list just for section_end flags, but that's probably not worth the complexity

Reply:
maybe we can kill section_end and use section_size instead, that will solve this problem. in fact, there should be no size either because flags are offset + size, so we must be able to just kill the _end thing and use the flag size when needed.

Reply:
that would apparently require complicated code if one wants section_end-related comments to appear in the disasm -- the naive way would be to check every section flag at every offset, but that sounds slow ... a faster way would be to have an ordered linked list or array of section_end offsets and maintain a marker for current disasm offset

Reply:
merge in the fi->section_end solution first?